src/k8s-extension/azext_k8s_extension/partner_extensions/EntraWorkloadIAM.py

```diff
@@ -65,24 +84,25 @@ def Create(self, cmd, client, resource_group_name, cluster_name, name, cluster_t
         scope_cluster = ScopeCluster(release_namespace=release_namespace)
         ext_scope = Scope(cluster=scope_cluster, namespace=None)
 
-        # Create new dictionary where the keys of the user settings are all lowercase (but leave the
-        # others alone in case they are specific settings that have to be passed to the Helm chart).
-        validated_settings = dict()
-        all_user_settings = [CONFIG_SETTINGS_USER_TRUST_DOMAIN, CONFIG_SETTINGS_USER_TENANT_ID,
-                             CONFIG_SETTINGS_USER_LOCAL_AUTHORITY, CONFIG_SETTINGS_USER_JOIN_TOKEN]
-        for key, value in configuration_settings.items():
-            if key.lower() in all_user_settings:
-                validated_settings[key.lower()] = value
-            else:
-                validated_settings[key] = value
-        config_settings = validated_settings
+        # Turn configuration setting keys into lowercase to make them case-insensitive
+        config_settings = settings_dict_to_lowercase(configuration_settings)
+        config_protected_settings = settings_dict_to_lowercase(configuration_protected_settings)
 
         # Get user configuration values and remove them from the dictionary so that they aren't
         # passed to the Helm chart
         trust_domain = config_settings.pop(CONFIG_SETTINGS_USER_TRUST_DOMAIN, None)
         tenant_id = config_settings.pop(CONFIG_SETTINGS_USER_TENANT_ID, None)
         local_authority = config_settings.pop(CONFIG_SETTINGS_USER_LOCAL_AUTHORITY, None)
+
+        # The join token can also be provided by the user. However, we must prevent them from
```

Comment:
I wonder whether we should stop allowing it being passed in as config_settings and only allowing config_protected_settings. Thoughts?